src/test/scala/mesosphere/marathon/core/task/tracker/InstanceTrackerFixture.scala

```diff
@@ -0,0 +1,56 @@
+package mesosphere.marathon
+package core.task.tracker
+
+import akka.Done
+import akka.actor.ActorSystem
+import mesosphere.marathon.core.condition.Condition
+import mesosphere.marathon.core.instance.{Goal, Instance}
+import mesosphere.marathon.core.instance.update.InstanceChangedEventsGenerator
+import mesosphere.marathon.core.task.tracker.InstanceTracker.InstancesBySpec
+import mesosphere.marathon.state.Timestamp
+import mesosphere.marathon.test.Mockito
+import org.mockito.Matchers
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait InstanceTrackerFixture extends Mockito with ScalaFutures {
+  def actorSystem: ActorSystem
+
+  class InstanceStubbingException(message: String) extends RuntimeException(message)
+
+  import scala.collection.mutable
+  val goalChangeMap: mutable.Map[Instance.Id, Goal] = mutable.Map.empty[Instance.Id, Goal]
+  val instanceTracker: InstanceTracker = mock[InstanceTracker]
+  instanceTracker.setGoal(any, any, any) answers { args =>
+    val instanceId = args(0).asInstanceOf[Instance.Id]
+    val maybeInstance = instanceTracker.get(instanceId).futureValue
+    maybeInstance.map { instance =>
+      val goal = args(1).asInstanceOf[Goal]
+      goalChangeMap.update(instanceId, goal)
+      sendKilled(instance, goal)
+      Future.successful(Done)
+    }.getOrElse {
+      Future.failed(throw new InstanceStubbingException(s"instance $instanceId is not stubbed"))
+    }
+  }
+
+  def populateInstances(instances: Seq[Instance]): Unit = {
```

Comment:
I'm not a big fan of such fixtures since they hide implementation details and flaws in the design. If it is not trivial to test parts of our production code we should refactor and simplify that instead of adding complexity to our test code.

Reply:
please propose how to do it differently bu pushing into the branch if you have an opinion wrt that...

Reply:
I agree with @jeschkies sentiment but refactoring that part would explode the scope of this PR. I think the ideal would be an instanceTracker(Fixture) that can be created with a given set of instances.

The alternative w/o any improvement or fixture would be 3-8 lines of manual stubbing/mocking in each test case.